go/eks-cluster/main.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+
+	eks "github.com/aws/aws-cdk-go/awscdk/awseks"
+	ec2 "github.com/aws/aws-cdk-go/awscdk/awsec2"
+	iam "github.com/aws/aws-cdk-go/awscdk/awsiam"
+	autoscaling "github.com/aws/aws-cdk-go/awscdk/awsautoscaling"
+	cdk "github.com/aws/aws-cdk-go/awscdk"
+	jsii "github.com/aws/jsii-runtime-go"
+
+)
+
+func main() {
+
+	app := cdk.NewApp(nil)
+	stack := cdk.NewStack(app, jsii.String("MyEKSCluster"), nil)
+
+	vpc := ec2.NewVpc(stack, jsii.String("EKSVpc"), nil) // Create a new VPC for our cluster
+
+	// IAM role for our EC2 worker nodes
+	workerRole := iam.NewRole(stack, jsii.String("EKSWorkerRole"), &iam.RoleProps{
+		AssumedBy: iam.NewServicePrincipal(jsii.String("ec2.amazonaws.com"), nil),
+	});
+
+	eksCluster := eks.NewCluster(stack, jsii.String("Cluster"), &eks.ClusterProps{
+		Vpc: vpc,
+		DefaultCapacity: jsii.Number(0), // we want to manage capacity our selves
+		Version: eks.KubernetesVersion_V1_16(),
+	})
+
+	onDemandASG := autoscaling.NewAutoScalingGroup(stack, jsii.String("OnDemandASG"), &autoscaling.AutoScalingGroupProps{
+		Vpc: vpc,
+		Role: workerRole,
+		MinCapacity: jsii.Number(1),
+		MaxCapacity: jsii.Number(10),
+		InstanceType: ec2.NewInstanceType(jsii.String("t3.medium")),
+		MachineImage: eks.NewEksOptimizedImage(&eks.EksOptimizedImageProps{
+			KubernetesVersion: jsii.String("1.14"),
+			NodeType: eks.NodeType_STANDARD,
+		}),
+		UpdateType: autoscaling.UpdateType_ROLLING_UPDATE,
+	});
+
+	eksCluster.ConnectAutoScalingGroupCapacity(onDemandASG, &eks.AutoScalingGroupOptions{});
```

Comment:
For some reason I cannot use `nil` here:

```console
❯ cdk synth                                                                       [01:12:48]
panic: "Got 'undefined' for non-optional instance of {\"docs\":{\"summary\":\"options for adding auto scaling groups, like customizing the bootstrap script.\"},\"name\":\"options\",\"type\":{\"fqn\":\"monocdk.aws_eks.AutoScalingGroupOptions\"}}"

goroutine 1 [running]:
github.com/aws/jsii-runtime-go.InvokeVoid(0x1698c00, 0xc0004b4d20, 0x16ba4c6, 0x1f, 0xc00006fe48, 0x2, 0x2)
```

Reply:
Scratch that. The problem is that the argument is indeed non optional, even though all its properties are. So it makes sense `nil` is not an allowed value.